### PR TITLE
Add payment form validation

### PR DESCRIPTION
**Pull request by douglashall**
Screen Shot 2019-06-19 at 3 30 56 PM
Screen Shot 2019-06-19 at 3 31 09 PM
Screen Shot 2019-06-19 at 3 31 29 PM
Screen Shot 2019-06-19 at 3 31 58 PM
Screen Shot 2019-06-19 at 3 32 13 PM

### CODE REVIEW

File: src/payment/PaymentForm.jsx [hunk 1]
```diff
@@ -1,27 +1,109 @@
...
+    // TODO: implement payment submission
+  };
+
+  validateCardDetails(cardNumber, securityCode, cardExpirationMonth, cardExpirationYear) {
+    const errors = {};
+
+    const { card, isValid } = CardValidator.number(cardNumber);
+    if (cardNumber) {
+      if (!isValid) {
+        errors.cardNumber = 'Invalid card number';
```
Comment (albemarle) [inline src/payment/PaymentForm.jsx]: Do we need to i18n these?

File: src/payment/PaymentForm.jsx [hunk 2]
```diff
@@ -1,27 +1,109 @@
...
+        }
+      }
+    }
+
+    const currentMonth = new Date().getMonth() + 1;
+    const currentYear = new Date().getFullYear();
+    if (
+      cardExpirationMonth &&
+      parseInt(cardExpirationMonth, 10) < currentMonth &&
+      parseInt(cardExpirationYear, 10) === currentYear
```
Comment (albemarle) [inline src/payment/PaymentForm.jsx]: What about a card that expired in, say, December 2018?

Comment (douglashall) [reply src/payment/PaymentForm.jsx]: The year dropdown populates with a range from current year to current year + 15. The user will not be able to select a year in the past.

**Comment by douglashall** [comment]
@albemarle Error messages are now i18n'ed. We good?

**Pull request merged**